src/Compiler/Microsoft.AspNetCore.Razor.Language/src/BoundAttributeDescriptor.cs

```diff
@@ -49,6 +50,19 @@ protected BoundAttributeDescriptor(string kind)
 
     public bool CaseSensitive { get; protected set; }
 
+    private bool? _isDirectiveAttribute;
```

Comment:
I'm not entirely sure that `bool?` is an atomic read/write, so I'm concerned about observing sheared values. Asking the runtime folks about this.

Reply:
Talking with Toub, it is possible for reads/write to be sheared, so we need to use a threadsafe read/write mechanism below.

Reply:
@333fred I'm not entirely up on what might/might not cause shearing so could I get you to confirm that my update is safe? If not I'll probably just abandon the change here since the effect is quite small.